build.rs

```diff
@@ -1,62 +1,76 @@
-    let antlr_path = "/home/rrevenantt/dev/antlr4/tool/target/antlr4-4.8-2-SNAPSHOT-complete.jar";
 
-    for (grammar, arg) in grammars.into_iter().zip(additional_args) {
-        //ignoring error because we do not need to run anything when deploying to crates.io
-        let _ = gen_for_grammar(grammar, antlr_path, arg);
+    fn antlr_path_env() -> String {
+        env::var(ANTLR_PATH).expect("Missing ANTLR_PATH environmental variable")
     }
 
-    println!("cargo:rerun-if-changed=build.rs");
+    pub fn generate_grammars() {
+        for (grammar, arg) in GRAMMARS_AND_ADDITION_ARGS {
+            gen_for_grammar(grammar, *arg);
+        }
 
-    println!("cargo:rerun-if-changed=/home/rrevenantt/dev/antlr4/tool/target/antlr4-4.8-2-SNAPSHOT-complete.jar");
-}
+        println!("cargo:rerun-if-changed=build.rs");
+
+        println!("cargo:rerun-if-env-changed={}", ANTLR_PATH);
 
-fn gen_for_grammar(
-    grammar_file_name: &str,
-    antlr_path: &str,
-    additional_arg: Option<&str>,
-) -> Result<(), Box<Error>> {
-    // let out_dir = env::var("OUT_DIR").unwrap();
-    // let dest_path = Path::new(&out_dir);
-
-    let input = env::current_dir().unwrap().join("grammars");
-    let file_name = grammar_file_name.to_owned() + ".g4";
-
-    let c = Command::new("java")
-        .current_dir(input)
-        .arg("-cp")
-        .arg(antlr_path)
-        .arg("org.antlr.v4.Tool")
-        .arg("-Dlanguage=Rust")
-        .arg("-o")
-        .arg("../tests/gen")
-        .arg(&file_name)
-        .args(additional_arg)
-        .spawn()
-        .expect("antlr tool failed to start")
-        .wait_with_output()?;
-    // .unwrap()
-    // .stdout;
-    // eprintln!("xx{}",String::from_utf8(x).unwrap());
-
-    println!("cargo:rerun-if-changed=grammars/{}", file_name);
-    Ok(())
+        println!("cargo:rerun-if-changed={}", antlr_path_env());
+    }
+
+    fn gen_for_grammar(grammar_file_name: &str, additional_arg: Option<&str>) {
+        let input = env::current_dir()
+            .expect("Could not retrieve current directory")
+            .join("grammars");
+        let file_name = grammar_file_name.to_owned() + ".g4";
+
+        let exit_status = Command::new("java")
+            .current_dir(input)
+            .arg("-cp")
+            .arg(antlr_path_env())
+            .arg("org.antlr.v4.Tool")
+            .arg("-Dlanguage=Rust")
+            .arg("-o")
+            .arg("../tests/gen")
+            .arg(&file_name)
+            .args(additional_arg)
+            .spawn()
+            .expect("antlr tool failed to start")
+            .wait()
+            .expect("antlr tool wasn't running");
+
+        assert!(
+            exit_status.success(),
+            "antlr tool failed to generate grammar"
+        );
+
+        let exit_status = Command::new("cargo")
+            .arg("fmt")
+            .spawn()
+            .expect("`cargo fmt` failed")
+            .wait()
+            .expect("rustfmt wasn't running");
```

Comment:
Running rustfmt isn't necessary for the the build to work, but it is a nice touch to get generated code which has been already formatted.